#! /bin/sh /usr/share/dpatch/dpatch-run
## 19_umask.dpatch by  <andete@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: use 0666 isof 0600 as umask when creating files

@DPATCH@
diff -urNad trunk~/dumpcap.c trunk/dumpcap.c
--- wireshark-1.2.0/dumpcap.c.orig	2009-06-30 22:15:25.000000000 -0700
+++ wireshark-1.2.0/dumpcap.c	2009-06-30 22:15:41.000000000 -0700
@@ -1883,7 +1883,7 @@
       } else {
         /* Try to open/create the specified file for use as a capture buffer. */
         *save_file_fd = ws_open(capfile_name, O_RDWR|O_BINARY|O_TRUNC|O_CREAT,
-                             0600);
+                             0666);
       }
     }
     is_tempfile = FALSE;
--- wireshark-1.2.0/file.c.orig	2009-06-30 22:17:26.000000000 -0700
+++ wireshark-1.2.0/file.c	2009-06-30 22:17:54.000000000 -0700
@@ -1206,7 +1206,7 @@
 
   if (*out_filenamep != NULL) {
     out_filename = *out_filenamep;
-    out_fd = ws_open(out_filename, O_CREAT|O_TRUNC|O_BINARY, 0600);
+    out_fd = ws_open(out_filename, O_CREAT|O_TRUNC|O_BINARY, 0666);
     if (out_fd == -1)
       open_err = errno;
   } else {
--- wireshark-1.2.0/ringbuffer.c.orig	2009-06-30 22:18:46.000000000 -0700
+++ wireshark-1.2.0/ringbuffer.c	2009-06-30 22:19:25.000000000 -0700
@@ -123,7 +123,7 @@
     return -1;
   }
 
-  rb_data.fd = ws_open(rfile->name, O_RDWR|O_BINARY|O_TRUNC|O_CREAT, 0600);
+  rb_data.fd = ws_open(rfile->name, O_RDWR|O_BINARY|O_TRUNC|O_CREAT, 0666);
 
   if (rb_data.fd == -1 && err != NULL) {
     *err = errno;
